### PR TITLE
chapter10_part4: /070_Index_Mgmt/20_Custom_Analyzers.asciidoc

### PR DESCRIPTION
**Pull request by tangmisi**

### CODE REVIEW

File: 070_Index_Mgmt/20_Custom_Analyzers.asciidoc [hunk 1]
```diff
@@ -1,74 +1,39 @@
...
 --
-Character filters((("character filters"))) are used to ``tidy up'' a string before it is tokenized.
-For instance, if our text is in HTML format, it will contain HTML tags like
-`<p>` or `<div>` that we don't want to be indexed. We can use the
-{ref}/analysis-htmlstrip-charfilter.html[`html_strip` character filter]
-to remove all HTML tags and to convert HTML entities like `&Aacute;` into the
-corresponding Unicode character `Á`.
-
-An analyzer may have zero or more character filters.
+字符过滤器 ((("character filters"))) 用来 `整理` 一个尚未被分词的字符串。例如，如果我们的文本是HTML格式的，它会包含像 `<p>` 或者 `<div>` 这样的HTML标签，这些标签是我们不想索引的。我们可以使用 {ref}/analysis-htmlstrip-charfilter.html[`html清除` 字符过滤器] 来移除掉所有的HTML标签，并且像把 `&Aacute;` 转换为相关的Unicode字符 `Á`  这样，转换HTML转义符。
```
Comment (leo650) [inline 070_Index_Mgmt/20_Custom_Analyzers.asciidoc]: 1.corresponding  是否翻译成  ”相对应的“比较好一点。
2.HTML entities  好像是”html 实体“，与”HTML转义符“是不是等同概念，需要了解一下。

File: 070_Index_Mgmt/20_Custom_Analyzers.asciidoc [hunk 2]
```diff
@@ -1,74 +1,39 @@
...
-{ref}/analysis-stop-tokenfilter.html[`stop` token filters],
-but there are many more available in Elasticsearch.
-{ref}/analysis-stemmer-tokenfilter.html[Stemming token filters]
-``stem'' words to ((("stemming token filters")))their root form. The
-{ref}/analysis-asciifolding-tokenfilter.html[`ascii_folding` filter]
-removes diacritics,((("ascii_folding filter"))) converting a term like `"très"` into `"tres"`. The
-{ref}/analysis-ngram-tokenfilter.html[`ngram`] and
-{ref}/analysis-edgengram-tokenfilter.html[`edge_ngram` token filters] can produce((("edge_engram token filter")))((("ngram and edge_ngram token filters")))
-tokens suitable for partial matching or autocomplete.
+经过分词，作为结果的词单元流会按照指定的顺序通过指定的词单元过滤器 (("token filters"))) 。
```
Comment (leo650) [inline 070_Index_Mgmt/20_Custom_Analyzers.asciidoc]: 少了斜体_token stream_

File: 070_Index_Mgmt/20_Custom_Analyzers.asciidoc [hunk 3]
```diff
@@ -1,74 +1,39 @@
...
-{ref}/analysis-stop-tokenfilter.html[`stop` token filters],
-but there are many more available in Elasticsearch.
-{ref}/analysis-stemmer-tokenfilter.html[Stemming token filters]
-``stem'' words to ((("stemming token filters")))their root form. The
-{ref}/analysis-asciifolding-tokenfilter.html[`ascii_folding` filter]
-removes diacritics,((("ascii_folding filter"))) converting a term like `"très"` into `"tres"`. The
-{ref}/analysis-ngram-tokenfilter.html[`ngram`] and
-{ref}/analysis-edgengram-tokenfilter.html[`edge_ngram` token filters] can produce((("edge_engram token filter")))((("ngram and edge_ngram token filters")))
-tokens suitable for partial matching or autocomplete.
+经过分词，作为结果的 _词单元流_ 会按照指定的顺序通过指定的词单元过滤器 (("token filters"))) 。
```
Comment (leo650) [inline 070_Index_Mgmt/20_Custom_Analyzers.asciidoc]: (("token filters"))) 前面少了一个括号。

**Comment by leo650** [comment]
LGTM

**Comment by Geolem** [comment]
LGTM

**Comment by medcl** [comment]
编译失败，
a2x: executing: "xmllint" --nonet --noout --valid "/Users/medcl/Documents/codes/github/book.xml"

/Users/medcl/Documents/codes/github/book.xml:8737: parser error : Opening and ending tag mismatch: literal line 8737 and ulink
erence/current/analysis-lowercase-tokenizer.html">&#8216;小写<literal></ulink>
                                                                               ^
/Users/medcl/Documents/codes/github/book.xml:8737: parser error : Opening and ending tag mismatch: ulink line 8737 and literal
ide/en/elasticsearch/reference/master/analysis-stop-tokenfilter.html"></literal>
                                                                               ^
/Users/medcl/Documents/codes/github/book.xml:22755: parser error : Opening and ending tag mismatch: indexterm line 22754 and tertiary
y><secondary>dictionary stemmers</secondary><tertiary>Hunspell stemmer</tertiary
                                                                               ^
/Users/medcl/Documents/codes/github/book.xml:22756: parser error : Opening and ending tag mismatch: simpara line 22749 and indexterm
</indexterm>
            ^
/Users/medcl/Documents/codes/github/book.xml:22766: parser error : Opening and ending tag mismatch: section line 22747 and simpara
other open and closed source projects.</simpara>
                                                ^
/Users/medcl/Documents/codes/github/book.xml:23097: parser error : Opening and ending tag mismatch: chapter line 22348 and section
</section>
          ^
/Users/medcl/Documents/codes/github/book.xml:23617: parser error : Opening and ending tag mismatch: part line 19538 and chapter
</chapter>
          ^
/Users/medcl/Documents/codes/github/book.xml:26261: parser error : Opening and ending tag mismatch: book line 6 and part
</part>
       ^
/Users/medcl/Documents/codes/github/book.xml:26262: parser error : Extra content at the end of the document
<part id="aggregations">

@Geolem  预览确认之后才能打 to be merge 标签，要谨慎一点

**Comment by tangmisi** [comment]
好像被我搞乱了，close掉重新开一个

File: 070_Index_Mgmt/20_Custom_Analyzers.asciidoc [hunk 4]
```diff
@@ -1,74 +1,39 @@
 [[custom-analyzers]]
-=== Custom Analyzers
+=== 自定义分词器
```
Comment (medcl) [inline 070_Index_Mgmt/20_Custom_Analyzers.asciidoc]: analyzer——》分析器

File: 070_Index_Mgmt/20_Custom_Analyzers.asciidoc [hunk 5]
```diff
@@ -1,74 +1,39 @@
...
 
-While Elasticsearch comes with a number of analyzers available out of the box,
-the real power comes from the ability to create your own custom analyzers
-by combining character filters, tokenizers, and token filters in a
-configuration that suits your particular data.
+虽然Elasticsearch带有一些现成的分析器，然而在分析器上Elasticsearch真正的强大之处在于，你可以通过在一个适合你的特定数据的设置之中组合字符过滤器、分词器、词汇单元过滤器来创建自定义的分析器。
 
-In <<analysis-intro>>, we said that an _analyzer_ is a wrapper that combines
-three functions into a single package,((("analyzers", "character filters, tokenizers, and token filters in"))) which are executed in sequence:
+在 <<analysis-intro>> 我们说过，一个 _分词器_ 就是在一个包里面组合了三种函数的一个包装器， ((("analyzers", "character filters, tokenizers, and token filters in"))) 三种函数按照顺序被执行:
```
Comment (medcl) [inline 070_Index_Mgmt/20_Custom_Analyzers.asciidoc]: 一个_ 分析器

**Comment by medcl** [comment]
LGTM

**Pull request merged**